From f4bd089b155ebb564d168cf752faf5a945b7d4a6 Mon Sep 17 00:00:00 2001
From: Mel Gorman <mgorman@techsingularity.net>
Date: Wed, 23 Mar 2022 14:47:19 +0000
Subject: [PATCH] rt: Add documentation describing what kernel debug options to
 add for testing

References: SLE Realtime Extension
Patch-mainline: Never, SLERT specific

Stress testing of the kernel may benefit from setting debugging options that
are too expensive to run under production but do not necessarily belong in
rt_debug either. Document the minimum recommended list of debug options.

Signed-off-by: Mel Gorman <mgorman@suse.de>
---
 Documentation/rt-debug-options | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/Documentation/rt-debug-options b/Documentation/rt-debug-options
new file mode 100644
index 000000000000..ecf631b10838
--- /dev/null
+++ b/Documentation/rt-debug-options
@@ -0,0 +1,16 @@
+set:CONFIG_CIFS_DEBUG2
+set:CONFIG_DEBUG_ATOMIC_SLEEP
+set:CONFIG_DEBUG_PREEMPT
+set:CONFIG_DEBUG_RODATA_TEST
+set:CONFIG_DEBUG_RT_MUTEXES
+set:CONFIG_DEBUG_SPINLOCK
+set:CONFIG_FSCACHE_DEBUG
+set:CONFIG_PROVE_LOCKING
+set:CONFIG_RCU_EQS_DEBUG
+set:CONFIG_SLUB_DEBUG_ON
+set:CONFIG_DEBUG_IRQFLAGS
+set:CONFIG_DEBUG_LIST
+set:CONFIG_DEBUG_PLIST
+set:CONFIG_DEBUG_VM
+set:CONFIG_DEBUG_PER_CPU_MAPS
+set:CONFIG_DEBUG_RWSEMS
